App/PyImports/Calculators/CryspyCalculator.py

```diff
@@ -24,29 +27,94 @@ def __init__(self, main_rcif_path, parent=None):
         # cryspy
         self._main_rcif_path = main_rcif_path
         self._main_rcif = None
+        self._phases_path = ""
+        self._experiments_path = ""
         self.final_chi_square = None
         self._cryspy_obj = self._createCryspyObj() #cryspy.rhochi_read_file(self._main_rcif_path)
         # project dict
         self._project_dict = {}
         self.setProjectDictFromCryspyObj()
 
-    def _createCryspyObj(self):
-        """Temporary solution to create cryspy object from separate rcif files"""
-        full_rcif_content = ''
+    def updatePhases(self, phases_path):
+        """
+        Parse the relevant phases file and update the corresponding model
+        """
+        self._phases_path = phases_path
+        rcif_content = ""
+
+        # This will read the CIF file
+        if os.path.isfile(self._phases_path):
+            with open(self._phases_path, 'r') as f:
+                phases_rcif_content = f.read()
+                rcif_content += phases_rcif_content
+
+        # find the name of the new phase
+        data_segment = phases_rcif_content.find('data_') # first instance only
+        data_segment_length = len('data_')
+        end_loc = data_segment + phases_rcif_content[data_segment:].find('\n')
+        new_phase_name = phases_rcif_content[data_segment+data_segment_length:end_loc].strip()
+
+        # This will replace phase name in EXPERIMENT
+        experiment_segment = self.replacePhaseInSegment(EXPERIMENT_SEGMENT, new_phase_name)
+
+        # This will remove unneeded 'global_'s in both segments
+        experiment_segment = experiment_segment.replace('global_', '')
+        rcif_content = rcif_content.replace('global_', '')
+
+        # Concatenate the old/corrected experiment info
+        rcif_content = experiment_segment + rcif_content
+
+        # This will update the CrysPy object
+        self._cryspy_obj = cryspy.RhoChi()
+        self._cryspy_obj.from_cif(rcif_content)
+
+        # This will re-create all local directories
+        self.setProjectDictFromCryspyObj()
+
+        # This will notify the GUI models changed
+        self.projectDictChanged.emit()
+
+    def replacePhaseInSegment(self, segment, new_phase_name):
+        """
+        Replaces original phase name with the given name in a segment
+        """
+        segment_content = self._parseSegment(segment)
+        old_phase_name = self._phase_name
```

Comment:
`self._phase_name` is not defined in init

Reply:
it is now